docs/src/components/NavBar.astro

```diff
@@ -0,0 +1,40 @@
+---
+import astroConfig from "../../astro.config.mjs";
+import { Image } from 'astro:assets';
+import logo from '../assets/logo-small.png';
+const homeUrl = `${astroConfig.base}`;
+const isaJsonSpecsUrl = `${astroConfig.base}/mars-isa-json`;
+---
+
+<nav class="relative dark:bg-gray-800 dark:border-gray-600 after:pointer-events-none after:absolute after:inset-x-0 after:bottom-0 after:h-px after:bg-white/10">
+  <div class="mx-auto max-w-7xl px-2 sm:px-6 lg:px-8">
+    <div class="relative flex h-16 items-center justify-between">
+      <div class="absolute inset-y-0 left-0 flex items-center sm:hidden">
+        <!-- Mobile menu button-->
+        <button type="button" command="--toggle" commandfor="mobile-menu" class="relative inline-flex items-center justify-center rounded-md p-2 text-gray-400 hover:bg-white/5 hover:text-white focus:outline-2 focus:-outline-offset-1 focus:outline-indigo-500">
+          <span class="absolute -inset-0.5"></span>
+          <span class="sr-only">Open main menu</span>
+          <Image src={logo} alt="Small MARS logo" style="width: 64px;">
+        </button>
+      </div>
+      <div class="flex flex-1 items-center justify-center sm:items-stretch sm:justify-start">
+        <div class="flex shrink-0 items-center">
+          <Image src={logo} alt="Small MARS logo" style="width: 64px;">
+        </div>
+        <div class="hidden sm:ml-6 sm:block">
+          <div class="flex space-x-4">
+            <a href={homeUrl} aria-current="page" class="rounded-md bg-gray-950/50 px-3 py-2 text-sm font-medium text-white">Home</a>
```

Comment:
Currently home has a active state, I would put a conditional that checks what the current page is and set the active state classes based on that